be/src/exec/vectorized/json_scanner.cpp

```diff
@@ -275,61 +304,336 @@ Status JsonReader::close() {
             }
-            rows_to_read--;
-            _next_line++;
+
+            RETURN_IF_ERROR(_filter_object_with_json_root(obj));
+
+            RETURN_IF_ERROR(_process_object(chunk, slot_descs, obj));
+        } else {
+            std::string err_msg = "Unsupported data type in json array";
+            _state->append_error_msg_to_file("", err_msg);
+            _counter->num_rows_filtered++;
+            return Status::DataQualityError(err_msg.c_str());
         }
-    } while (rows_to_read > 0);
+    }
+    return Status::OK();
+}
+
+Status JsonReader::_process_array_with_json_path(Chunk* chunk, const std::vector<SlotDescriptor*>& slot_descs,
+                                                 simdjson::ondemand::array& arr) {
+    for (auto a : arr) {
+        if (a.error()) {
+            std::string err_msg = strings::Substitute("Failed to iterate json array with jsonpath. code=$0, error=$1",
+                                                      a.error(), simdjson::error_message(a.error()));
+            _state->append_error_msg_to_file("", err_msg);
+            _counter->num_rows_filtered++;
+            return Status::DataQualityError(err_msg.c_str());
+        }
+
+        simdjson::ondemand::json_type tp;
+        auto err = a.type().get(tp);
+        if (err) {
+            std::string err_msg =
+                    strings::Substitute("Failed to determine type in json array with jsonpath. code=$0, error=$1", err,
+                                        simdjson::error_message(err));
+            _state->append_error_msg_to_file("", err_msg);
+            _counter->num_rows_filtered++;
+            return Status::DataQualityError(err_msg.c_str());
+        }
+
+        if (tp == simdjson::ondemand::json_type::object) {
+            simdjson::ondemand::object obj;
+            auto err = a.get_object().get(obj);
+            if (err) {
+                std::string err_msg =
+                        strings::Substitute("Failed to parse json in array with jsonpath. code=$0, error=$1", a.error(),
+                                            simdjson::error_message(a.error()));
+                _state->append_error_msg_to_file("", err_msg);
+                _counter->num_rows_filtered++;
+                return Status::DataQualityError(err_msg.c_str());
+            }
+
+            RETURN_IF_ERROR(_filter_object_with_json_root(obj));
+
+            RETURN_IF_ERROR(_process_object_with_json_path(chunk, slot_descs, obj));
+        } else {
+            std::string err_msg = "Unsupported data type in json array";
+            _state->append_error_msg_to_file("", err_msg);
+            _counter->num_rows_filtered++;
+            return Status::DataQualityError(err_msg.c_str());
+        }
+    }
+    return Status::OK();
+}
+
+Status JsonReader::_process_object(Chunk* chunk, const std::vector<SlotDescriptor*>& slot_descs,
+                                   simdjson::ondemand::object& obj) {
```

Comment:
```suggestion
                                   simdjson::ondemand::object* obj) {
```

Reply:
In fact, the object is an iterator and would not be modified explicitly, so it cannot be declared with const.Is the name of function _process_xxx misleading ?